data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/PipelineConnector.java

```diff
@@ -54,6 +57,32 @@ public void stop() {
     public void output(final Collection<T> records) {
         if (buffer != null && !isStopRequested.get()) {
             for (T record : records) {
+		if(record.getData() instanceof JacksonSpan) {
+		    try {
+		        final JacksonSpan span = (JacksonSpan)record.getData();
+			JacksonSpan newSpanEvent = JacksonSpan.builder()
+			          .withJsonData(span.toJsonString())
```

Comment:
This should be `withData(recordEvent.toMap())` similar to line 76 below.

Reply:
As it is not obvious, the `toMap` method returns a converted value. So it is not returning the same `Map` object as is help by the original `Event`.